### PR TITLE
fix: fix the parsing of the error string in some edge cases

### PR DESCRIPTION
**Pull request by commoddity**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -514,25 +510,22 @@ func parseErrorResponse(errResponse *http.Response) error {
...
 	if err != nil {
-		return err
+		return errString
 	}
 
 	var errorMap map[string]string
 	err = json.Unmarshal(body, &errorMap)
 	if err != nil {
-		return err
+		return errString
```
Comment (ezeike) [inline client/client.go:523]: The actual error from io.ReadAll and json.Unmarshal are being lost with this change returning errString.

Comment (commoddity) [reply client/client.go:523]: Yes that's intended.

Comment (commoddity) [reply client/client.go:523]: If either of those errors out it means there was an issue with the response body itself (which  contains a more detailed error message in production cases). In that case we still want to get at the very least an error message containing a the status code and status text.

Comment (commoddity) [reply client/client.go:523]: That said, this won't actually happen in production as in actual usage of PHD the errors are always sent using the `utils-go` client that adds a detailed error message to the response body.

This PR is mainly just to fix an issue that was happening when using the client in unit tests where the response body is missing the body.

**Review by ezeike — APPROVED** [review]

**Pull request merged**